gen3/bin/kube-setup-aurora-monitoring.sh

```diff
@@ -0,0 +1,169 @@
+    SECURITY DEFINER;
+SQL
+
+  gen3_log_info "Succesfully added the function and schema"
+}
+
+if [ $# -eq 0 ]; then
+    echo "Error: No argument provided. You must provide the name of the Aurora cluster to operate against"
+    exit 1
+fi
+
+get_all_dbs databaseArray
+
+# Loop through every database, creating the schema and function
+for db in "${databaseArray[@]}"
+do
+  create_schema_and_function $db
+done
+
+
+# Set up the agent
+#==============================
+
+# Get the instances in the Aurora cluster
+  # We'll take the name of the cluster as the first argument, so we won't need to go digging for that. Instead, we'll just
+  # pull out connection strings and ports for each instance
+
+instances=$(aws rds describe-db-instances --filters "Name=db-cluster-id,Values=$1" --no-paginate | jq '.DBInstances[].Endpoint.Address,.DBInstances[].Endpoint.Port' | tr -d '"')
+clusterEndpoint=$(aws rds describe-db-cluster-endpoints --db-cluster-identifier "$1" | jq ' .DBClusterEndpoints[0].Endpoint' | tr -d '"')
+
+postgresString=""
+for instance in "${instances[@]}" 
+do
+  instanceArray=($instance)
+  datadogUserPassword=$(jq --arg instance "$clusterEndpoint" '.[$instance].datadog_db_password' $(gen3_secrets_folder)/datadog/datadog_db_users.json | tr -d '"')
+  postgresString+=$(cat /home/aidan/cloud-automation/kube/services/datadog/postgres.yaml | yq --arg url ${instanceArray[0]} --yaml-output '.instances[0].host = $url' | yq --arg password $datadogUserPassword --yaml-output '.instances[0].password = $password')
+done
+
+confd=$(yq -n --yaml-output --arg postgres "$postgresString" '.clusterAgent.confd."postgres.yaml" = $postgres | .clusterChecksRunner.enabled = true')
+
+#We'll need two ways to do this, one for commons where Datadog is managed by ArgoCD, and another for commons where 
+#it's directly installed
+
+if kubectl get applications.argoproj.io -n argocd datadog-application &> /dev/null
+then
+  gen3_log_info "We detected an ArgoCD application named 'datadog-application,' so we're modifying that"
+
+  patch=$(yq -n --yaml-output --arg confd "$confd" '.spec.source.helm.values = $confd')
+  
+  echo "$patch" > /tmp/confd.yaml
+
+  kubectl patch applications.argoproj.io datadog-application --type merge -n argocd --patch-file /tmp/confd.yaml
+
+else
+  gen3_log_info "We didn't detect an ArgoCD application named 'datadog-application,' so we're going to reinstall the DD Helm chart"
+  
+  (cat kube/services/datadog/values.yaml | yq --arg endpoints "$postgresString" --yaml-output '.clusterAgent.confd."postgres.yaml" = $endpoints | .clusterChecksRunner.enabled = true') > $(gen3_secrets_folder)/datadog/datadog_values.yaml
+  helm repo add datadog https://helm.datadoghq.com --force-update 2> >(grep -v 'This is insecure' >&2)
+  helm repo update 2> >(grep -v 'This is insecure' >&2)
+  helm upgrade --install datadog -f "$(gen3_secrets_folder)/datadog/datadog_values.yaml" datadog/datadog -n datadog --version 3.6.4 2> >(grep -v 'This is insecure' >&2)
```

Comment:
Should we just call kube-setup-datadog, so that we only have one place where we manage pinned DD versions and changes?

Reply:
We totally could, but that would require changing how kube-setup-datadog works, since this is referring the values file in a different place. I have a story out to change how we handle secrets/sensitive info for Helm charts as a whole, I think we should probably discuss that as a whole team to decide how far we want to tear apart code we're interested in deprecating.